File: src/main/java/net/lomeli/lomlib/core/command/CommandClearEntities.java
package net.lomeli.lomlib.core.command;

import java.util.List;

import net.minecraft.command.CommandException;
import net.minecraft.command.ICommandSender;
import net.minecraft.entity.Entity;
import net.minecraft.entity.EntityLivingBase;

public class CommandClearEntities extends CommandBaseLomLib {
    @Override
    public String getCommandName() {
        return "clear-entities";
    }

    @Override
    public int getRequiredPermissionLevel() {
        return 2;
    }

    @Override
    public void processCommand(ICommandSender sender, String[] args) throws CommandException {
        if (sender == null) return;
        List<?> entityList = sender.getEntityWorld().loadedEntityList;
        if (entityList == null || entityList.isEmpty()) return;
        try {
            for (Object entity : entityList) {
                if (entity instanceof Entity) {
                    if (!(entity instanceof EntityLivingBase))
                        ((Entity) entity).setDead();
                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        sendMessage(sender, "command.lomlib.clear-entities.success");
    }
}

File: src/main/kotlin/net/lomeli/lomlib/core/command/CommandClearEntities.kt
package net.lomeli.lomlib.core.command

import net.minecraft.command.CommandException
import net.minecraft.command.ICommandSender
import net.minecraft.entity.Entity
import net.minecraft.entity.EntityLivingBase
import net.minecraft.server.MinecraftServer

class CommandClearEntities : CommandBaseLomLib() {
    override fun getCommandName(): String = "clear-entities"

    override fun getRequiredPermissionLevel(): Int = 2

    @Throws(CommandException::class)
    override fun execute(server: MinecraftServer, sender: ICommandSender, args: Array<String>) {
        if (sender == null) return
        val entityList = sender.entityWorld.loadedEntityList
        if (entityList == null || entityList.isEmpty()) return
        try {
            for (entity in entityList) {
                if (entity is Entity) {
                    if (entity !is EntityLivingBase)
                        entity.setDead()
                }
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }

        sendMessage(sender, "command.lomlib.clear-entities.success")
    }


}